project/web/pages/reset_password_form.py

```diff
@@ -0,0 +1,41 @@
+import time
+from typing import Union
+
+from selenium.webdriver.common.by import By
+
+from common._webdriver_qa_api.core.base_elements import BaseElement
+from common._webdriver_qa_api.web.web_elements import WebElement, WebTextBox
+from common.facade import logger
+from .sign_in_page import SignInPage
+
+
+class ResetPasswordForm(SignInPage):
+
+    def __init__(self, should_present=True):
+        super().__init__(should_present=False)
+        self.locator_type = By.XPATH
+        self.locator = "//form[@action='/users/password']"
+        self.name = "Reset password Form"
+
+        self.lbl_form_title = WebElement(By.XPATH, "//div[@class='sign-form']/h3")
+        self.txb_username = WebTextBox(By.XPATH, self.locator + "//input[@id='user_username']")
+        self.btn_submit = WebTextBox(By.ID, self.locator + "//input[@type='submit']")
+        if should_present:
+            self.assert_page_present(30)
+
+    def is_page_present(self, second: Union[int, float] = 0.1) -> bool:
```

Comment:
No, it's a good sample how user can override core layer behavior for their goals. 
In this case for verify that form present we need to check that element is visible for user instead of checking that element attached to DOM

Reply:
Agree. Thanks for the explanation